src/index.ts

```diff
@@ -0,0 +1,323 @@
+        //   provider
+        // );
+        const abi = RrpBeaconServer.abi;
+        const rrpBeaconServer = new ethers.Contract(
+          chain.contracts.RrpBeaconServer,
+          abi,
+          provider
+        );
+
+        // **************************************************************************
+        // 3. Run each keeper job
+        // **************************************************************************
+        for (const {
+          templateId,
+          oisTitle,
+          endpointName,
+          deviationPercentage,
+          keeperSponsor,
+          requestSponsor,
+        } of triggers.rrpBeaconServerKeeperJobs) {
+          // **************************************************************************
+          // 4. Fetch template by ID
+          // **************************************************************************
+          console.log("[DEBUG]\tfetching template...");
+          const template = await airnodeRrp.templates(templateId);
+          if (!template) {
+            console.log("[ERROR]\ttemplate not found");
+          }
+
+          // **************************************************************************
+          // 5. Make API request
+          // **************************************************************************
+          console.log("[DEBUG]\tmaking API request...");
+          const configOis = ois.find((o) => o.title === oisTitle)!;
+          const configEndpoint = configOis.endpoints.find(
+            (e) => e.name === endpointName
+          )!;
+          const templateParameters = safeDecode(template.parameters);
+          const reservedParameters = getReservedParameters(
+            configEndpoint,
+            templateParameters || {}
+          );
+          if (!reservedParameters._type) {
+            console.log("[ERROR]\treserved parameter 'type' is missing");
+            return;
+          }
+          const sanitizedParameters: adapter.Parameters = removeKeys(
+            templateParameters || {},
+            RESERVED_PARAMETERS
+          );
+          const adapterApiCredentials = apiCredentials
+            .filter((c) => c.oisTitle === oisTitle)
+            .map((c) => removeKey(c, "oisTitle") as adapter.ApiCredentials);
+
+          const options: adapter.BuildRequestOptions = {
+            ois: configOis,
+            endpointName,
+            parameters: sanitizedParameters,
+            metadataParameters: {}, // TODO: https://github.com/api3dao/airnode/pull/697
+            apiCredentials: adapterApiCredentials,
+          };
+
+          const apiResponse = await adapter.buildAndExecuteRequest(options);
+          if (!apiResponse || !apiResponse.data) {
+            console.log("[ERROR]\tfailed to fetch data from API");
+            return;
+          }
+          console.log("[INFO]\tAPI server response data:", apiResponse.data);
+          // TODO: should we really return here or 0 could be a valid response?
```

Comment:
🤔 I think I added this check because I wanted to prevent dividing by zero [here](https://github.com/api3dao/airkeeper/blob/384766263f9378e8d9a65a4ddc2da4e304a8c996/src/index.ts#L172) but you are probably right and I'll re-think this.